docs/adrs/0002-field-for-data-type-form-information.md

```diff
@@ -0,0 +1,162 @@
+# 1. Field for Data Type / Form Information
+
+Follows: https://mitlibraries.github.io/guides/misc/adr.html
+
+Date: 2024-02-15
+
+## Status
+
+Proposed
+
+## Context
+
+### Need
+
+Identify a pre-existing, or new, TIMDEX field to store information about GIS datasets that will support required facet filters, while also potentially serving other TIMDEX data sources.
+
+### Overview
+During a project that introduced geospatial data into TIMDEX, two new sources were added `gisogm` and `gismit`, both of which use the new Transmogrifier transformer 
+`MITAardvark`. Records for these sources come from the [GeoHarvester](https://github.com/MITLibraries/geo-harvester) which produces [Aardvark metadata schema](https://opengeometadata.org/ogm-aardvark/) JSON records.    
+
+During work on a new geospatial TIMDEX UI, the question was raised as to what fields should be used for two new required facet filters to mirror functionality in the [current Geoweb application](https://geodata.mit.edu/):
+
+  * "Data Type"
+    * values like: `[Polygon, Point, Raster, Image, etc.]`
+  * "Format"
+    * values like: `[Shapefile, TIFF, GeoTIFF, JPEG, etc.]`
+
+This revealed a potential gap in the TIMDEX data model.  While the TIMDEX field `format` is a great fit for the "Format" values we expect to see in that facet filter, there does not appear to be 
+an appropriate field to hold values that would populate the "Data Type" facet filter.
+
+It is worth noting that the facet filter "Content Types", driven by the TIMDEX field `content_type`, is set statically for these records as `Geospatial data`.  This allows them to sit next to 
+other TIMDEX records with `content_type` values like `[Language material, Musical sound recording, Thesis, etc.]`, thereby also removing `content_type` as a possible field for the issue at hand. 
+
+Analysis of the source `MITAardvark` records suggests that the Aardvark field `gbl_resourceType_sm` contains the values we'll want for the facet filter "Data Type".  The 
+Aardvark schema defines this field as:
+
+> "...categories for classifying the spatial type or structure of a dataset"
+
+In this way, the TIMDEX data model is lacking a field that can be used to describe the "spatial type or structure of a dataset".  Maybe stated more generally, a field that describes
+the structure of the data _inside_ the format noted in `format`.
+
+It is also worth noting that data values for both facets -- present in fields `dct_format_s` and `gbl_resourceType_sm` from the Aardvark records -- will be controlled values
```

Comment:
I wonder if `ResourceType` is more or less confusing than `Form` for the new object name? I _do_ think `Form` works and I will support it, but the fact that at least one other use case of this type of data refers to it as `ResourceType` seems worth explicitly considering.

Reply:
@JPrevost - do you think `ResourceType` could still take values formerly used in `literary_form`? e.g.

```json
"resource_type":[
  {
    "value":"Nonfiction",
    "kind":"Literary Form"
  }
]
```

I'd also wonder how this would to `content_type` which we [define in the spec](https://mitlibraries.github.io/timdex/reference/#definition-Record) as:

> "Type of content of item; e.g., "still image", "text", etc."

Sketching an example:

```json
{
  "content_type": "Geospatial data",   # this feels pretty well established
  "format": "Shapefile",
  "resource_type": [
    {
      "value":"Polygon",
      "kind": "Data Type"
    },
    {
      "value":"Vector",
      "kind": "Data Type"
    }
]
}
```

Given the GIS data, it seems that whatever we name this field it should allow multiple values (some MIT and OGM records have 3-4 "Data Type" qualities in the metadata).  I'd worry that `content_type` and `resource_type` seem conceptually similar, but `content_type` is single value string, while `resource_type` is multivalued object.  And, that `format` kind of sits between them hierarchically.

Reply:
I think `literary_form` could still collapse into this object with either the name `form` or `resourceType`.  I'm not sure which I prefer yet fwiw. I think `form` has a meaning in our current language that is not how we are using it here. I also think that we aren't using it _wrong_ fwiw, but it isn't the first meaning I would intuit when trying to use the data.

Reply:
Also noting that our current `content_type` appears to be an array of Strings and not just a single String.

Reply:
Also nothing that our mappings for our sources should be confirmed.

We have thesis records from Alma and DSpace for the same thesis that have different content_types, literary_forms, and formats.

This is not directly related to if we create a new object, but the lack of consistency across sources feels important to note as we consider what these closely related fields _mean_.

Example:
```json
{
  "source": "DSpace@MIT",
  "title": "Gardens of resistance",
  "contributors": [
    {
      "kind": "author",
      "value": "Jhaveri, Nynika\n            (Nynika P.)"
    },
    {
      "kind": "other",
      "value": "Massachusetts Institute of Technology. Department of Architecture."
    },
    {
      "kind": "department",
      "value": "Massachusetts Institute of Technology. Department of Architecture"
    }
  ],
  "contentType": [
    "Thesis"
  ],
  "literaryForm": null,
  "format": "electronic resource"
},
{
  "source": "MIT Alma",
  "title": "Gardens of resistance",
  "contributors": [
    {
      "kind": "Not specified",
      "value": "Jhaveri, Nynika (Nynika P.)"
    },
    {
      "kind": "Not specified",
      "value": "Massachusetts Institute of Technology. Department of Architecture"
    }
  ],
  "contentType": [
    "Manuscript language material"
  ],
  "literaryForm": "Nonfiction",
  "format": null
}
```

Reply:
> Also noting that our current `content_type` appears to be an array of Strings and not just a single String.

Good catch!  Thanks.  Crossing that out from above.